core/src/main/java/org/elasticsearch/action/search/RemoteClusterService.java

```diff
@@ -203,25 +207,30 @@ boolean isRemoteNodeConnected(final String remoteCluster, final DiscoveryNode no
      */
     Map<String, List<String>> groupClusterIndices(String[] requestIndices, Predicate<String> indexExists) {
         Map<String, List<String>> perClusterIndices = new HashMap<>();
+        Set<String> remoteClusterNames = this.remoteClusters.keySet();
         for (String index : requestIndices) {
             int i = index.indexOf(REMOTE_CLUSTER_INDEX_SEPARATOR);
-            String indexName = index;
-            String clusterName = LOCAL_CLUSTER_GROUP_KEY;
             if (i >= 0) {
                 String remoteClusterName = index.substring(0, i);
-                if (isRemoteClusterRegistered(remoteClusterName)) {
+                List<String> clusters = clusterNameResolver.resolveClusterNames(remoteClusterNames, remoteClusterName);
```

Comment:
can we maybe return an emtpy list instead of null, I think we don't necessarily need the null invariant?